### ISSUE TITLE
Docs: Embedding IPython

### ISSUE THREAD

**Issue by olenz** [ipython/ipython#5887]
The docs on [Embedding IPython](https://github.com/ipython/ipython/blob/master/docs/source/interactive/reference.rst#embedding-ipython) state about the embedded shell: "Note however that any changes you make to values while in the shell do not propagate back to the running code, so it is safe to modify your values because you won't break your code in bizarre ways by doing so."

My tests show that this is not true. When I call the following minimal script:

```
from __future__ import print_function
import IPython
a = 42
IPython.embed()
print("a={}".format(a))
```

And execute "a=23" in the embedded shell, then the program will output "a=23".

Do I misunderstand the docs, or what is wrong?

**Comment by lesley2958** [comment]
working on the documentation change!

**Comment by jzf2101** [comment]
@minrk do you mean this only to be a documentation change and not a change to the underlying behavior?

**Comment by minrk** [comment]
That’s the immediate issue, yes.

**Comment by takluyver** [comment]
Fixed by #10891
